doc/source/glossary.rst

```diff
@@ -0,0 +1,188 @@
+    a pointer.  Pointers are usually the same length as the maximum address of the address space, since they
+    should be able to point to any address within the space.
+
+R
+-
+.. _Range:
+
+Range
+    This is the set of the possible output values for a mapping or mathematical function.
+
+S
+-
+.. _Struct:
+
+Struct, Structure
+    A means of containing multiple different :ref:`type<types>` associated together.  A struct typically contains
+    other :ref:`type<types>`, usually :ref:`aligned<alignment>` (unless :ref:`packing<packed>` is involved).  In this way
+    the :ref:`members<member>` of a type can be accessed by finding the data at the relative :ref:`offset<offset>` to
+    the start of the structure.
+
+.. _Symbol:
+
+Symbol
+    This is used in many different contexts, as a short term for many things.  Within Volatility, a symbol is a
+    construct that usually encompasses a specific type :ref:`type<Type>` at a specfific :ref:`offset<Offset>`,
+    representing a particular instance of that type within the memory of a compiled and running program.  An example
+    would be the location in memory of a list of active tcp endpoints maintained by the networking stack
+    within an operating system.
+
+T
+-
+.. _Template:
+
+Template
+    Within volatility 3, the term template applies to a :ref:`type<Type>` that has not yet been instantiated or linked
+    to any data or a specific location within memory.  Once a type has been tied to a particular chunk of data, it is
+    called an :ref:`object<Object>`.
+
+.. _Translation Layer:
+
+Translation Layer
+    This is a specific type of :ref:`data layer<Data Layer>`, a non-contiguous group of bytes that can be references by
+    a unique :ref:`offset<Offset>` within the layer.  In particular, translation layers translates (or :ref:`maps<Map>`)
+    requests made of it to a location within a lower layer.  This can be either linear (a one-to-one mapping between bytes)
+    or non-linear (a group of bytes :ref:`maps<Map>` to a larger or smaller group of bytes.
+
+.. _Type:
+
+Type
+    This is a structure definition of multiple elements that expresses how data is laid out.  Basic types define how
+    the data should be interpretted in terms of a run of bits (or more commonly a collection of 8 bits at a time,
+    called bytes).  More complex types can be made up of other types combined together at specific locations known
+    as :ref:`structs<Struct>` or repeated, known as :ref:`array<Array>`.  They can even defined types at the same
+    location depending on the data itself, known as :ref:`Unions<Union>`.  Once a type has been linked to a specific
+    chunk of data, the result is referred to as an :ref:`object<object>`.
+
+U
+-
+.. _Union:
+
+Union
+    A union is a type that can have can hold multiple different subtypes, which specifically overlap.  A union is means
```

Comment:
Suggesting: 
In essence, a union is an alias for multiple types. Just like a struct, it has multiple members, but the bytes assigned to a union will only ever hold the value of one member, the one that was last written to. Depending on what data a union is holding, only a subset of its members will point to accurate data. (Assuming the data wasn't corrupted in the first place) To illustrate: Picture a box that can only ever have one thing inside. The box is suitable for a set of items (the union members): Apple, melon and cherry. The box will be big enough to hold a melon, but even though multiple apples and a lot of cherries could fit in the spot of one melon, our union will only ever hold a single apple, cherry or melon.

Reply:
Mmmm, I'm not keen on the illustration, here.  It doesn't feel as though it clarifies matters and I also don't want to potentially talk down to people.  I've added a rewording that hopefully gets across what we want to communicate better without having to resort to metaphors or similes which don't quite fit...